### DISCUSSION TITLE
DeepSpeed Integration

### DISCUSSION THREAD

**Discussion by ramonpzg** [outlines-dev/outlines#289] [Q&A]
Hello -- I was wondering if it is possible to integrate outlines with `deepspeed` or `deepspeed-mii`? If so, are there any examples that could be shared? If not, are there any plans to have the two work well with each other?

Many thanks! :)

**Comment by brandonwillard** [comment]
Can you give an example of how you would imagine using `deepspeed` with `outlines`?  For example, are you considering using `outlines` within an inference workflow that follows something like [this in the tutorial](https://www.deepspeed.ai/tutorials/inference-tutorial/#end-to-end-gpt-neo-27b-inference)?
